src/Packagist/WebBundle/Controller/ApiController.php

```diff
@@ -57,6 +57,36 @@ public function packagesAction(Request $req)
     }
 
     /**
+     * @Route("/api/create-package", name="generic_create", defaults={"_format" = "json"})
+     * @Method({"POST"})
+     */
+    public function createPackageAction(Request $request)
+    {
+
+        $payload = json_decode($request->getContent(), true);
+        if (!$payload) {
+            return new JsonResponse(array('status' => 'error', 'message' => 'Missing payload parameter'), 406);
+        }
+        $url = $payload['repository']['url'];
```

Comment:
do we actually need the ``repository`` part here ? I suggest using ``$payload['url']`` to be simpler